[BE] Fix flake8 B027 errors - missing abstractmethod decorator

Skylion007 opened the pull request:
Enables B027 and applies fixes by adding abstract method decorators. Autofix generated by ruff master.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

Skylion007 committed:
[BE]: Apply B027 fixes

Skylion007 committed:
Add noqa

Skylion007 committed:
Update ruff to 0.0.264

Skylion007 committed:
Revert errant change

Skylion007 committed:
Remove false positives and get rid of empty __init__ in ABCs

Skylion007 committed:
Update ruff to latest

ezyang commented:
This isn't an autofix though, you need to audit each of these no? Can you comment in why things were determined to be abstractmethod or not

Skylion007 commented:
> This isn't an autofix though, you need to audit each of these no? Can you comment in why things were determined to be abstractmethod or not

I actually autofixed all of the errors and then removed any that caused the test suite to fail (by adding noqa ignores). I could have instead added blank functions to all the subclasses of these ABC classes were it was intentional, but that seemed like a pain for extending new classes on methods are essentially hooks for abc methods so I just added ignores in. 

It doesn't make any sense to define an `__init__(self)` on an ABC since you cannot instantiate it anyway, although definiting an abstract `__init__` method that takes certain args (like the node ctor), can be useful. So I removed all the blank `__init__(self)` from abc classes since they were not doing anything useful.

ezyang commented:
@pytorchbot merge
